client/hostinfo_unix.cpp

```diff
@@ -181,6 +181,13 @@ extern "C" {
 #define LINUX_LIKE_SYSTEM 1
 #endif
 
+#if WASM
+    #include <emscripten.h>
```

Comment:
Please move this include to the top of the file together with other includes (keeping the `#if` guard of course)

Reply:
I separate the include from function definition. It cannot be on top. It must be after Linux definitions.